substrafl/evaluation_strategy.py

```diff
@@ -145,22 +139,55 @@ def _check_rounds_consistency(self, num_rounds: Optional[int] = None):
                 exceeds num_rounds. Defaults to None.
 
         Raises:
-            ValueError: num_rounds must not be smaller than self._rounds if self._rounds is an int
-            ValueError: num_rounds must not be be smaller than the largest value of self._rounds
+            ValueError: num_rounds must not be smaller than self._eval_frequency
+            ValueError: num_rounds must not be be smaller than the largest value of self._eval_rounds
         """
 
-        if isinstance(self._rounds, int) and num_rounds < self._rounds:
+        if self._eval_frequency is not None and num_rounds < self._eval_frequency:
             raise ValueError(
-                f"Not possible to test every {self._rounds} rounds (from EvaluationStrategy) "
+                f"Not possible to test every {self._eval_frequency} rounds (from EvaluationStrategy) "
                 f"as total number of rounds (num_rounds) from the Strategy is {num_rounds}."
             )
-        elif isinstance(self._rounds, list) and max(self._rounds) > num_rounds:
+
+        if self._eval_rounds is not None and max(self._eval_rounds) > num_rounds:
             # rounds is a list of round indices
             raise ValueError(
                 f"Not possible to test rounds (from EvaluationStrategy.rounds) greater than {num_rounds}"
                 "(num_rounds) from the Strategy."
             )
 
+    def _check_eval_frequency(self):
+        """Check eval_frequency validity.
+
+        Raises:
+            ValueError: eval_frequency must be a positive int
+            TypeError: eval_frequency must be of type int or None
+        """
+        if isinstance(self._eval_frequency, int):
+            if self._eval_frequency <= 0:
+                raise ValueError(f"eval_frequency must be positive, eval_frequency={self._eval_frequency} found")
+        elif self._eval_frequency is not None:
+            raise TypeError(f"eval_frequency must be of type int, {type(self._eval_frequency)} found")
+
+    def _check_eval_rounds(self):
+        """Check eval_rounds validity.
+
+        Raises:
+            ValueError: if eval_rounds is a list it cannot be empty
+            TypeError: if eval_rounds is a list it must be filled with variables of type int
+            ValueError: if eval_rounds is a list it must be filled with positive integers
+            TypeError: eval_rounds must be of type list or None
+        """
+        if isinstance(self._eval_rounds, list):
+            if not self._eval_rounds:
+                raise ValueError(f"eval_rounds cannot be empty, eval_rounds={self._eval_rounds} found")
+            elif not all(isinstance(r, int) for r in self._eval_rounds):
+                raise TypeError(f"eval_rounds must be of type list of ints, {type(self._eval_rounds)} found")
```

Comment:
Nitpick: isn't `type(self._eval_rounds)` always going to be `list`? Maybe we could have a more helpful error message:

```suggestion
            elif not all(isinstance(r, int) for r in self._eval_rounds):
                raise TypeError(f"eval_rounds must be of type list of ints, {next(type(x) for x in self._eval_rounds if not isinstance(x, int))} found")
```

(by the way, maybe we could test for `collections.abc.Iterable` rather than just `list`)

Reply:
Thanks for the catch, I agree :) 
I print all the list in case there is more than one element with the wrong type :) .